cmd/go-tpc/main.go

```diff
@@ -93,18 +100,31 @@ func newDB(targets []string, driver string, user string, password string, dbName
 	hash.Write([]byte(password))
 	hash.Write([]byte(dbName))
 	hash.Write([]byte(connParams))
+
+	if driver == mysqlDriver && len(sslCA) > 0 {
+		registerMysqlTLSConfig()
+	}
+
 	for i, addr := range targets {
 		hash.Write([]byte(addr))
 		switch driver {
 		case mysqlDriver:
+			var tlsName string = "preferred"
+			if len(sslCA) > 0 {
```

Comment:
```suggestion
			if len(sslCA) > 0 || len(sslKey) > 0 || len(sslCert) > 0 {
```

Reply:
Refer https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#enable-authentication

> By default, the server-to-client authentication is optional. Even if the client does not present its certificate of identification during the TLS handshake, the TLS connection can be still established.

If sslKey and sslCert are not provided, we will only authenticate the TiDB server from the MySQL client.

Reply:
I think valid configurations are:
- Only `sslCA`  provided
- `sslCA`, `sslCert`, `sslKey` all provided.

Not valid:
- `sslCert` and/or `sslKey` set, but no `sslCA`.
- `sslCert` set, but not `sslKey`
- `sslKey` set, but not `sslCert`

So I think we should set the `tlsName` to custom if any of `sslCert`, `sslKey` or `sslCA` is set and then later on verify if it is valid (which might already be done)